### PR TITLE
Add request id to logging

### PR DESCRIPTION
**Pull request by dfinteligenz**
## Description

- Add Request ID to all logs in the Go SDK
- Update demo Go processes

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other changes (ci configuration, documentation or any other kind of changes)

## Checklist:

- [x] I have created tests for my code changes, and the tests are passing.
- [ ] I have executed the pre-commit hooks locally.
- [ ] My change requires a change to the documentation (create a new issue if the documentation has not been updated).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: demo/exit/go.mod [hunk 1]
```diff
@@ -33,8 +33,11 @@ require (
...
 	golang.org/x/crypto v0.9.0 // indirect
-	golang.org/x/sys v0.8.0 // indirect
+	golang.org/x/exp v0.0.0-20230905200255-921286631fa9 // indirect
+	golang.org/x/sys v0.12.0 // indirect
 	golang.org/x/text v0.9.0 // indirect
 	gopkg.in/ini.v1 v1.67.0 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
+
+replace github.com/konstellation-io/kai-sdk/go-sdk => ../../go-sdk
```
Comment (JYisus) [inline demo/exit/go.mod:43]: Don't forget to delete this before merge

File: demo/cronjob-trigger/main.go [hunk 2]
```diff
@@ -49,20 +50,20 @@ func cronjobRunner(tr *trigger.Runner, sdk sdk.KaiSDK) {
...
 
 	c := cron.New(
-		cron.WithLogger(sdk.Logger),
+		cron.WithLogger(sdk.Logger.WithName("[CRONJOB]")),
 	)
 
 	_, err := c.AddFunc("@every 30s", func() {
 		requestID := uuid.New().String()
-		sdk.Logger.Info("Cronjob triggered, new message sent", "requestID", requestID)
+		sdk.Logger.Info("Cronjob triggered, new message sent", kaiconstants.LoggerRequestID, requestID)
```
Comment (JYisus) [inline demo/cronjob-trigger/main.go:58]: IMO we should not expose these constants to sdk users, they may be helpful for sdk devs, but users  should be responsible of what they log. Maybe the internal package is a good place for them.

**Review by JYisus — CHANGES_REQUESTED** [review]

File: demo/exit/go.mod (hunk 1 above)
Comment (dfinteligenz) [reply demo/exit/go.mod:43]: This is for testing the SDK change in the local environment without having to setup a full KAI locally, and they will be removed in the near future, so I'd let them as they are.

File: demo/cronjob-trigger/main.go (hunk 2 above)
Comment (dfinteligenz) [reply demo/cronjob-trigger/main.go:58]: We need to expose them for trigger processes where the requestID has not been set in the logger context yet, we can think of another way of doing it, but the requestID is a must to have full execution tracking

File: go-sdk/sdk/kaiconstants/kai_constants.go [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package kaiconstants
```
[the remark this answers is not in the document]
Comment (dfinteligenz) [reply go-sdk/sdk/kaiconstants/kai_constants.go:1]: Agreed, we could move them to the KAI package.

**Review by kafkaphoenix — DISMISSED** [review]

**Review by alexrodfe — DISMISSED** [review]

**Review by kafkaphoenix — DISMISSED** [review]

**Review by alexrodfe — DISMISSED** [review]

**Review by alexrodfe — APPROVED** [review]

**Review by kafkaphoenix — APPROVED** [review]

**Review by alexrodfe — DISMISSED** [review]

**Review by kafkaphoenix — DISMISSED** [review]

**Review by kafkaphoenix — APPROVED** [review]

**Review by kafkaphoenix — APPROVED** [review]

**Pull request merged**